src/cpu/pred/BranchPredictor.py

```
@@ -1063,16 +1063,32 @@ class BTBTAGE(TimedBaseBTBPredictor):
-    histLengths = [16]
-    numDelay = 0
+class MicroTAGE(TimedBaseBTBPredictor):
+    """Micro-sized BTB TAGE predictor used alongside uBTB"""
+    type = 'MicroTAGE'
+    cxx_class = 'gem5::branch_prediction::btb_pred::MicroTAGE'
+    cxx_header = "cpu/pred/btb/microtage.hh"
+
+    needMoreHistories = Param.Bool(True, "MicroTAGE needs more histories")
+    updateOnRead = Param.Bool(False,"Enable update on read, no need to save tage meta in FTQ")
+    # Keep vector parameters consistent with numPredictors to avoid constructor asserts.
+    numPredictors = Param.Unsigned(1, "Number of TAGE predictors")
```

Ruff (0.14.13)

`Param` may be undefined, or defined from star imports

(F405)